### PR TITLE
Arachne 2696 files naming convention

### PR DESCRIPTION
**Pull request by YaroslavTir**
the new format for analysis archives  
CODE-name_of_analysis_DATETIME.zip

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java [hunk 1]
```diff
@@ -244,38 +251,31 @@ public AnalysisFile saveFileByLink(String link, IUser user, A analysis, String l
...
+            ZipUtil.addZipEntries(zos, allFilesWithoutSqlFile);
+
+            String analysisName = getAnalysisName(genericSqlFile);
+            String fileName = AnalysisArchiveUtils.getArchiveFileName(CommonAnalysisType.COHORT, analysisName);
+            MultipartFile sqlArchive = new MockMultipartFile(fileName, fileName, com.google.common.net.MediaType.ZIP.toString(),
+                    out.toByteArray());
             saveFile(sqlArchive, user, analysis, fileName, false, dataReference);
-        } catch (AlreadyExistException e) {
-            log.error("Failed to save file", e);
+
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: There are some behavior changes during refactoring. Now it tries to save output stream before zip stream actually closed. Looks suspicious and might be the reason of some unexpected errors.

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java [hunk 2]
```diff
@@ -244,38 +251,31 @@ public AnalysisFile saveFileByLink(String link, IUser user, A analysis, String l
...
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
-        StringBuilder generatedFileNameBuilder = new StringBuilder();
-        generatedFileNameBuilder.append(CommonAnalysisType.COHORT.getTitle());
-        try (final ZipOutputStream zos = new ZipOutputStream(out)) {
-            files.stream()
+    public void saveCohortAnalysisArchive(A analysis, DataReference dataReference, IUser user, List<MultipartFile> files) {
+
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream();
+             ZipOutputStream zos = new ZipOutputStream(out)) {
+            MultipartFile genericSqlFile = files.stream()
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: Now the whole collection handled with two loops. Not sure this is better than single-phase handling.

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java [hunk 3]
```diff
@@ -299,20 +299,22 @@ public String updateAnalysisFromMetaFiles(A analysis, List<MultipartFile> entity
...
         String baseName = FilenameUtils.getBaseName(file.getOriginalFilename());
         String extension = FilenameUtils.getExtension(file.getOriginalFilename());
-        DBMSType[] dbTypes = new DBMSType[]{DBMSType.POSTGRESQL, DBMSType.ORACLE, DBMSType.MS_SQL_SERVER,
-                DBMSType.REDSHIFT, DBMSType.PDW};
-        for (final DBMSType dialect : dbTypes) {
-            final String sql = SqlTranslate.translateSql(renderedSql, dialect.getOhdsiDB());
-            final String fileName = baseName + "."
-                    + dialect.getLabel().replace(" ", "-")
-                    + "." + extension;
+        List<DBMSType> dbTypes = Arrays.asList(POSTGRESQL, ORACLE, MS_SQL_SERVER, REDSHIFT, PDW);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: Why we should restrict translations only for presented DBMS? I've found the previous behavior with the same, but?

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java (hunk 1 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: hmm, this one is good, I would think about this

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java (hunk 2 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: I rewrote it in a more readable way(at least I think so)  I don't really like it when we iterate over the collection and do more that one thing at the time.

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java (hunk 3 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: have no idea, for some reason, we get only these particular dialects. I think it was discussed with the customer. I would generate SQL for each DBVSType, but need to get approved for this.

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java (hunk 1 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: That is interesting. According theByteArrayOutputStream jdoc, the close method does not affect anything: 

From ByteArrayOutputStream class:
```    
/**
     * Closing a <tt>ByteArrayOutputStream</tt> has no effect. The methods in
     * this class can be called after the stream has been closed without
     * generating an <tt>IOException</tt>.
     */
    public void close() throws IOException {
    }

```

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: Well, but my concern is more about `ZipOutputStream`

Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: fixed

File: src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java (hunk 3 above)
Comment (YaroslavTir) [reply src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/AnalysisFilesSavingServiceImpl.java]: fixed

**Pull request merged**